reana-ui/src/pages/workflowDetails/components/filePreview/FilePreview.js

```diff
@@ -0,0 +1,213 @@
+
+import client, { WORKFLOW_FILE_URL } from "~/client";
+import { getMimeType } from "~/util";
+
+import styles from "./FilePreview.module.scss";
+
+// ROOTPreview is lazily loaded to enable code splitting, so that jsroot is not part of
+// the main application bundle
+const ROOTPreview = lazy(() => import("./ROOTPreview.js"));
+
+/**
+ * Preview of image files.
+ */
+function ImagePreview({ fileName, url }) {
+  return (
+    <Modal.Content scrolling>
+      <img src={url} alt={fileName} />
+    </Modal.Content>
+  );
+}
+
+/**
+ * Preview of HTML files.
+ */
+function HTMLPreview({ url }) {
+  return (
+    <Modal.Content scrolling>
+      <Message
+        icon="info circle"
+        info
+        content={
+          <div className={styles["html-message"]}>
+            <span>Visualise this HTML file in a different tab.</span>
+            <Button
+              icon="external"
+              as="a"
+              href={url}
+              target="_blank"
+              rel="noopener noreferrer"
+              content="Open"
+              primary
+            />
+          </div>
+        }
+      />
+    </Modal.Content>
+  );
+}
+
+/**
+ * Preview of plain-text files.
+ */
+function TextPreview({ workflow, fileName }) {
+  const [content, setContent] = useState(null);
+
+  useEffect(() => {
+    client.getWorkflowFile(workflow, fileName).then((res) => {
+      let result = res.data;
+      if (typeof result === "object") {
+        result = JSON.stringify(result);
+      }
+      setContent(result);
+    });
+  }, [workflow, fileName]);
+
+  return (
+    <Modal.Content scrolling>
+      <pre>{content}</pre>
+    </Modal.Content>
+  );
+}
+
+/**
+ * Preview of PDF files.
+ */
+function PDFPreview({ fileName, url }) {
+  return (
+    <Modal.Content className={styles["fill-modal"]}>
+      <object data={url}>
+        <Message icon info style={{ margin: "1.5rem", width: "auto" }}>
+          <Icon name="info circle" />
+          <Message.Content>
+            Click{" "}
+            <a href={url} target="_blank" rel="noopener noreferrer">
+              {fileName}
+            </a>{" "}
+            to open the PDF file, or use the download button.
+          </Message.Content>
+        </Message>
+      </object>
+    </Modal.Content>
+  );
+}
+
+const PREVIEW_MIME_PREFIX_WHITELIST = {
+  "image/": ImagePreview,
+  "text/html": HTMLPreview,
+  "text/": TextPreview,
+  "application/json": TextPreview,
+  "application/pdf": PDFPreview,
+  "application/x-root": ROOTPreview,
+};
+
+const FILE_SIZE_LIMIT = 5 * 1024 ** 2; // 5MB
+
+/**
+ * Check if the given file name matches any given mime-type
+ * @param {Array} list Array of mime-types to check against
+ * @param {String} fileName File name to check
+ * @return {Boolean} Extension that matches the file name
+ */
+function matchesMimeType(list, fileName) {
+  const mimeType = getMimeType(fileName);
+  return mimeType && list.find((ext) => mimeType.startsWith(ext));
```

Comment:
```suggestion
  return (
    mimeType &&
    (list.find((ext) => mimeType === ext) ||
      list.find((ext) => mimeType.startsWith(ext)))
  );
```

What do you think about this approach? In this way, exact matches in the MIME types have "priority" over partial matches, regardless of the order in the `PREVIEW_MIME_PREFIX_WHITELIST` map (e.g. if `"text/"` comes before `"text/html"`.

Reply:
I have updated this to return the longest matching "mime-type prefix"